agent/config/runtime.go

```diff
@@ -651,7 +652,7 @@ type RuntimeConfig struct {
 	// EnableDebug is used to enable various debugging features.
 	//
 	// hcl: enable_debug = (true|false)
-	EnableDebug bool
+	EnableDebug atomic.Bool
```

Comment:
I think this should stay as a bool as this `RuntimeConfig` data structure could be copied and copying an atomic have unwanted side effects.

instead we will need to apply the config to the agent every time it's updated. You can add an atomic [here](https://github.com/hashicorp/consul-enterprise/blob/9fe5041517c9e9944268b18a567137580b41bcdb/agent/agent.go#L218-L218) in the agent struct and init that atomic with the value in the agent [Start](https://github.com/hashicorp/consul-enterprise/blob/9fe5041517c9e9944268b18a567137580b41bcdb/agent/agent.go#L582-L582) and everytime the config is [reloaded](https://github.com/hashicorp/consul-enterprise/blob/9fe5041517c9e9944268b18a567137580b41bcdb/agent/agent.go#L4114-L4114) apply it to the atomic.
Then you can use the atomic safely in the http handler.